File: speckit-chat-vscode/src/service/git-helper.ts
import { execFileSync, execSync } from 'child_process';

export function currentBranch(basePath: string, fallback = 'main'): string {
    try {
        const output = execSync('git rev-parse --abbrev-ref HEAD', {
            cwd: basePath,
            encoding: 'utf-8',
            timeout: 10_000,
        }).trim();
        return output || fallback;
    } catch {
        return fallback;
    }
}

export function gitAdd(basePath: string, relativePath: string): void {
    try {
        execFileSync('git', ['add', relativePath], {
            cwd: basePath,
            encoding: 'utf-8',
            timeout: 5_000,
        });
    } catch {
        // Best-effort — don't block if git isn't available
    }
}

File: speckit-chat-plugin/src/main/kotlin/com/speckit/plugin/service/GitHelper.kt
package com.speckit.plugin.service

import com.intellij.openapi.application.ApplicationManager
import java.io.File
import java.util.concurrent.TimeUnit

object GitHelper {

    fun currentBranch(basePath: String, fallback: String = "main"): String {
        return try {
            val process = ProcessBuilder("git", "rev-parse", "--abbrev-ref", "HEAD")
                .directory(File(basePath))
                .redirectErrorStream(true)
                .start()
            val output = process.inputStream.bufferedReader().readText().trim()
            if (process.waitFor() == 0) output else fallback
        } catch (_: Exception) { fallback }
    }

    fun gitAdd(basePath: String, relativePath: String) {
        ApplicationManager.getApplication().executeOnPooledThread {
            try {
                ProcessBuilder("git", "add", relativePath)
                    .directory(File(basePath))
                    .redirectErrorStream(true)
                    .start()
                    .waitFor(5, TimeUnit.SECONDS)
            } catch (_: Exception) {
                // Best-effort — don't block the UI if git isn't available
            }
        }
    }
}
